### PR TITLE
Add slack webhook endpoint to Api

### PR DESCRIPTION
**Pull request by ceddlyburge**
This PR has quite a few changes, as not all of the instructions in the readme were working. I had made some changes to make one thing work, and had unknowingly broken other things.

This is a weak point of the current setup / CI, so in theory at the moment you should walk through the instructions in the readme before merging to check that all 3 of the use cases work (running locally, running on a local k8s cluster and running on a hosted k8s cluster). This is probably impractical though, so we probably just have to be careful and accept the odd problem for the moment. At some point we will probably sort out a release pipeline, which should ensure that everything works on a hosted k8s cluster (and at this point, it working hosted would be the most crucial thing).

This code is currently running on a cluster on my personal google cloud subscription, and is integrated with the red badger slack. You can test it by doing `/get_manager @cedd.burge` in slack, or using curl below (assuming you are in the root of the repo).

```
curl -X POST -H "Content-Type: application/x-www-form-urlencoded" --data @lib/slack/exampleRequest.txt  http://34.105.155.126/v1.0/invoke/slack.slack/method/getManager
```

The api currently returns David Wynne as the manager no matter who is passed in, so hopefully there are no privacy concerns.

To test the code (as opposed to the deployment) please run through the instructions in the readme for one or all of use cases.

### CODE REVIEW

File: lib/slack/src/index.ts [hunk 1]
```diff
@@ -5,13 +5,75 @@ const app = express()
...
+//   team_domain: 'redbadger',
+//   channel_id: 'D01R6L9NQ3S',
+//   channel_name: 'directmessage',
+//   user_id: 'U01QP0VJWJG',
+//   user_name: 'cedd.burge',
+//   command: '/get_manager',
+//   text: '@cedd.burge',
+//   api_app_id: 'A01SHH2QF8S',
+//   is_enterprise_install: 'false',
+//   response_url: 'https://hooks.slack.com/commands/T02A80YKK/2001166095842/CZprfzCQr6UoWSc5d0DiZivb',
```
Comment (alsuren) [inline lib/slack/src/index.ts:36]: Fingers crossed there's nothing secret in here (or you scrambled it sufficiently).

**Review by alsuren — APPROVED** [review]
Looking good.

File: lib/slack/src/index.ts (hunk 1 above)
Comment (ceddlyburge) [reply lib/slack/src/index.ts:36]: Hi David, this is a good point, and these details are repeated in exampleRequest.txt.

I think there is nothing sensitive, but probably another pair of eyes would be useful.
 - The token is deprecated and we don't use it (and won't)
 - The response_url and similar are short lived, so are no longer valid
 - The api_app_id, team_id, channel_id and user_id are real things, but I don't think they are sensitive ([signed secrets](https://api.slack.com/authentication/verifying-requests-from-slack) are how we should verify that a message is genuinely coming from slack, when we get round to doing that)

What do you think? Potentially I can recreate some of these id's if needs be.

Comment (alsuren) [reply lib/slack/src/index.ts:36]: The main thing I was worried about was response_url, but if it's short-lived then that's fine. If `token` is also not usable for anything then I think we're fine.

**Pull request merged**